File: php/features.php
<?php
declare(strict_types=1);

// NoAsAServiceTwo SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class NoAsAServiceTwoFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new NoAsAServiceTwoBaseFeature();
            case "test":
                return new NoAsAServiceTwoTestFeature();
            default:
                return new NoAsAServiceTwoBaseFeature();
        }
    }
}

File: py/features.py
# NoAsAServiceTwo SDK feature factory

from feature.base_feature import NoAsAServiceTwoBaseFeature
from feature.test_feature import NoAsAServiceTwoTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NoAsAServiceTwoBaseFeature(),
        "test": lambda: NoAsAServiceTwoTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
